### PR TITLE
Add client-side commands

### PR DESCRIPTION
**Pull request by mattn**
See https://github.com/prabirshrestha/vim-lsp/pull/735

### CODE REVIEW

File: settings/rust-analyzer.vim [hunk 1]
```diff
@@ -11,4 +11,28 @@ augroup vimlsp_settings_rust-analyzer
       \ 'workspace_config': lsp_settings#get('rust-analyzer', 'workspace_config', {}),
       \ 'semantic_highlight': lsp_settings#get('rust-analyzer', 'semantic_highlight', {}),
       \ }
+  autocmd User lsp_setup call s:register_command()
 augroup END
+
+function! s:rust_analyzer_apply_source_change(context)
+    let l:command = get(a:context, 'command', {})
+
+    let l:workspace_edit = get(l:command['arguments'][0], 'workspaceEdit', {})
```
Comment (prabirshrestha) [inline settings/rust-analyzer.vim]: Would be good to check if arguments is not empty.

File: settings/eclipse-jdt-ls.vim [hunk 2]
```diff
@@ -11,4 +11,19 @@ augroup vimlsp_settings_eclipse_jdt_ls
       \ 'workspace_config': lsp_settings#get('eclipse-jdt-ls', 'workspace_config', {}),
       \ 'semantic_highlight': lsp_settings#get('eclipse-jdt-ls', 'semantic_highlight', {}),
       \ }
+  autocmd User lsp_setup call s:register_command()
 augroup END
+
+function! s:eclipse_jdt_ls_java_apply_workspaceEdit(context)
+    let l:command = get(a:context, 'command', {})
+    call lsp#utils#workspace_edit#apply_workspace_edit(l:command['arguments'][0])
```
Comment (prabirshrestha) [inline settings/eclipse-jdt-ls.vim]: Check if arguments not empty?

File: settings/eclipse-jdt-ls.vim [hunk 3]
```diff
@@ -11,4 +11,19 @@ augroup vimlsp_settings_eclipse_jdt_ls
...
+endfunction
+
+function! s:register_command()
+  augroup vimlsp_settings_eclipse_jdt_ls
+    au!
+  augroup END
+  if exists('*lsp#register_command')
```
Comment (prabirshrestha) [inline settings/eclipse-jdt-ls.vim]: If vim-lsp PR is checked in this wouldn’t be required.

File: settings/rust-analyzer.vim [hunk 4]
```diff
@@ -11,4 +11,28 @@ augroup vimlsp_settings_rust-analyzer
...
+    if !empty(l:workspace_edit)
+        call lsp#utils#workspace_edit#apply_workspace_edit(l:workspace_edit)
+    endif
+
+    let l:cursor_position = get(l:command['arguments'][0], 'cursorPosition', {})
+    if !empty(l:cursor_position)
+        call cursor(lsp#utils#position#_lsp_to_vim('%', l:cursor_position))
```
Comment (prabirshrestha) [inline settings/rust-analyzer.vim]: We should make lsp_to_vim public and document it.

File: settings/eclipse-jdt-ls.vim (hunk 3 above)
Comment (mattn) [reply settings/eclipse-jdt-ls.vim]: Thanks. But someone users older version. I will remove this check after few weeks goes.

**Pull request merged**